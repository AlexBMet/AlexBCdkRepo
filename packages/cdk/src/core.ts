export const generateResourcePrefix = (uniquePrefix: string, environment: string, region: string) => {
	return `${uniquePrefix}-${environment}-${region}`
};
