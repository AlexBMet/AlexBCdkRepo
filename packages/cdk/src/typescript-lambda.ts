import {Code, Function, Runtime, Tracing} from '@aws-cdk/aws-lambda';
import {LogGroup, RetentionDays} from '@aws-cdk/aws-logs';
import {CfnParameter, Construct, Duration, RemovalPolicy, Stack, Tag} from '@aws-cdk/core';

export class TypescriptLambda extends Stack {

	constructor(scope: Construct, id: string, props: {}) {
		super(scope, id, props);

		const STACK = Stack.of(this);
		const ENVIRONMENT = new CfnParameter(this, 'Environment', {
			allowedValues: ['dev', 'ci', 'stg', 'prod'],
			description: 'TBC',
			type: 'String'
		});
		const SERVICE_CODE = new CfnParameter(this, 'ServiceCode', {
			type: 'String',
			description: 'TBC'
		});
		const SERVICE_NAME = new CfnParameter(this, 'ServiceName', {
			type: 'String',
			description: 'TBC'
		});
		const SERVICE_OWNER = new CfnParameter(this, 'ServiceOwner', {
			type: 'String',
			description: 'TBC'
		});
		const BUCKET_NAME = new CfnParameter(this, 'SourceBucketName', {
			description: 'TBC',
			type: 'String'
		});
		const OBJECT_KEY = new CfnParameter(this, 'SourceObjectKey', {
			description: 'TBC',
			type: 'String'
		});
		const UNIQUE_PREFIX = new CfnParameter(this, 'UniquePrefix', {
			type: 'String',
			description: 'TBC'
		});
		const PREFIX = `${UNIQUE_PREFIX.value}-${ENVIRONMENT.value}-${STACK.region}`;

		Tag.add(this, 'Environment', `${ENVIRONMENT.value}`);
		Tag.add(this, 'ServiceCode', `${SERVICE_CODE.value}`);
		Tag.add(this, 'ServiceName', `${SERVICE_NAME.value}`);
		Tag.add(this, 'ServiceOwner', `${SERVICE_OWNER.value}`);

		// tslint:disable-next-line:function-constructor
		const lambdaFunction = new Function(this, 'TypeScriptLambda', {
			code: Code.fromCfnParameters({
				bucketNameParam: BUCKET_NAME,
				objectKeyParam: OBJECT_KEY
			}),
			description: 'TBC',
			functionName: `${PREFIX}-typescript-lambda`,
			handler: 'dist/index.handler',
			memorySize: 128,
			runtime: Runtime.NODEJS_12_X,
			timeout: Duration.seconds(30),
			tracing: Tracing.ACTIVE
		});

		new LogGroup(this, 'TypeScriptLambdaLogs', {
			logGroupName: `/aws/lambda/${lambdaFunction.functionName}`,
			retention: RetentionDays.THREE_DAYS,
			removalPolicy: RemovalPolicy.DESTROY
		});
	}
}
