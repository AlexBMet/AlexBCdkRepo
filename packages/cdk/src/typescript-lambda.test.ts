import {SynthUtils} from '@aws-cdk/assert'; // tslint:disable-line:no-implicit-dependencies
import {Stack} from '@aws-cdk/core';
import {TypescriptLambda} from './typescript-lambda';

describe('[typescript-lambda.ts] unit tests', () => {

	describe('[stack]', () => {
		it('must create an lambda function', async () => {
			const stack = new TypescriptLambda(new Stack(), 'TypescriptLambda', {});
			expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
		});
	});
});
