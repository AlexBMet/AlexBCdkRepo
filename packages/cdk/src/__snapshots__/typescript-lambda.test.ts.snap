// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[typescript-lambda.ts] unit tests [stack] must create an lambda function 1`] = `
Object {
  "Parameters": Object {
    "Environment": Object {
      "AllowedValues": Array [
        "dev",
        "ci",
        "stg",
        "prod",
      ],
      "Description": "TBC",
      "Type": "String",
    },
    "ServiceCode": Object {
      "Description": "TBC",
      "Type": "String",
    },
    "ServiceName": Object {
      "Description": "TBC",
      "Type": "String",
    },
    "ServiceOwner": Object {
      "Description": "TBC",
      "Type": "String",
    },
    "SourceBucketName": Object {
      "Description": "TBC",
      "Type": "String",
    },
    "SourceObjectKey": Object {
      "Description": "TBC",
      "Type": "String",
    },
    "UniquePrefix": Object {
      "Description": "TBC",
      "Type": "String",
    },
  },
  "Resources": Object {
    "TypeScriptLambdaE9A80D1B": Object {
      "DependsOn": Array [
        "TypeScriptLambdaServiceRoleDefaultPolicyC84F6F7E",
        "TypeScriptLambdaServiceRole5A9D8B00",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Ref": "SourceBucketName",
          },
          "S3Key": Object {
            "Ref": "SourceObjectKey",
          },
        },
        "Description": "TBC",
        "FunctionName": Object {
          "Fn::Join": Array [
            "",
            Array [
              Object {
                "Ref": "UniquePrefix",
              },
              "-",
              Object {
                "Ref": "Environment",
              },
              "-",
              Object {
                "Ref": "AWS::Region",
              },
              "-typescript-lambda",
            ],
          ],
        },
        "Handler": "dist/index.handler",
        "MemorySize": 128,
        "Role": Object {
          "Fn::GetAtt": Array [
            "TypeScriptLambdaServiceRole5A9D8B00",
            "Arn",
          ],
        },
        "Runtime": "nodejs12.x",
        "Tags": Array [
          Object {
            "Key": "Environment",
            "Value": Object {
              "Ref": "Environment",
            },
          },
          Object {
            "Key": "ServiceCode",
            "Value": Object {
              "Ref": "ServiceCode",
            },
          },
          Object {
            "Key": "ServiceName",
            "Value": Object {
              "Ref": "ServiceName",
            },
          },
          Object {
            "Key": "ServiceOwner",
            "Value": Object {
              "Ref": "ServiceOwner",
            },
          },
        ],
        "Timeout": 30,
        "TracingConfig": Object {
          "Mode": "Active",
        },
      },
      "Type": "AWS::Lambda::Function",
    },
    "TypeScriptLambdaLogs9E207222": Object {
      "DeletionPolicy": "Delete",
      "Properties": Object {
        "LogGroupName": Object {
          "Fn::Join": Array [
            "",
            Array [
              "/aws/lambda/",
              Object {
                "Ref": "TypeScriptLambdaE9A80D1B",
              },
            ],
          ],
        },
        "RetentionInDays": 3,
      },
      "Type": "AWS::Logs::LogGroup",
      "UpdateReplacePolicy": "Delete",
    },
    "TypeScriptLambdaServiceRole5A9D8B00": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
        "Tags": Array [
          Object {
            "Key": "Environment",
            "Value": Object {
              "Ref": "Environment",
            },
          },
          Object {
            "Key": "ServiceCode",
            "Value": Object {
              "Ref": "ServiceCode",
            },
          },
          Object {
            "Key": "ServiceName",
            "Value": Object {
              "Ref": "ServiceName",
            },
          },
          Object {
            "Key": "ServiceOwner",
            "Value": Object {
              "Ref": "ServiceOwner",
            },
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "TypeScriptLambdaServiceRoleDefaultPolicyC84F6F7E": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "xray:PutTraceSegments",
                "xray:PutTelemetryRecords",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "TypeScriptLambdaServiceRoleDefaultPolicyC84F6F7E",
        "Roles": Array [
          Object {
            "Ref": "TypeScriptLambdaServiceRole5A9D8B00",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
  },
}
`;
